cpp/unit/ability.h

```diff
@@ -42,17 +43,14 @@ using ability_set = std::bitset<16>;
 using ability_id_t = unsigned int;
```

Comment:
how do you know that 16 is large enough for `ability_set`? you should add a member `ability_type::MAX`, and use that. also, ability_type should be an `enum class`. As is, we have int constants like `::openage::move` flying around in the global namespace.